apps/mark-scan/backend/src/custom-paper-handler/application_driver.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * This file is the layer between libs/custom-paper-handler/driver and mark-scan state machine.
+ * It abstracts application-specific logic but doesn't send low-level USB commands directly to
+ * the paper handler. Functions that are more complicated than a single driver command
+ * invocation belong here.
+ * A good rule of thumb is: if only the paper handler driver and external VX libs are being called,
+ * the function belongs here. If the state machine state is being read or updated, it belongs in
+ * ./state_machine.ts
+ */
+
+export function isPaperInInput(
+  paperHandlerStatus: PaperHandlerStatus
+): boolean {
+  return (
+    paperHandlerStatus.paperInputLeftInnerSensor ||
+    paperHandlerStatus.paperInputLeftOuterSensor ||
+    paperHandlerStatus.paperInputRightInnerSensor ||
+    paperHandlerStatus.paperInputRightOuterSensor
+  );
+}
+
+export function isPaperReadyToLoad(
+  paperHandlerStatus: PaperHandlerStatus
+): boolean {
+  return (
+    paperHandlerStatus.paperInputLeftInnerSensor &&
+    paperHandlerStatus.paperInputLeftOuterSensor &&
+    paperHandlerStatus.paperInputRightInnerSensor &&
+    paperHandlerStatus.paperInputRightOuterSensor
+  );
+}
+
+export function isPaperInScanner(
+  paperHandlerStatus: PaperHandlerStatus
+): boolean {
+  return (
+    paperHandlerStatus.paperPreCisSensor ||
+    paperHandlerStatus.paperPostCisSensor ||
+    paperHandlerStatus.preHeadSensor ||
+    paperHandlerStatus.paperOutSensor ||
+    paperHandlerStatus.parkSensor ||
+    paperHandlerStatus.paperJam ||
+    paperHandlerStatus.scanInProgress
+  );
+}
+
+export function isPaperInOutput(
+  paperHandlerStatus: PaperHandlerStatus
+): boolean {
+  return paperHandlerStatus.ticketPresentInOutput;
+}
+
+export function isPaperAnywhere(
+  paperHandlerStatus: PaperHandlerStatus
+): boolean {
+  return (
+    isPaperInInput(paperHandlerStatus) || isPaperInScanner(paperHandlerStatus)
+  );
+}
+
+export async function logRawStatus(driver: PaperHandlerDriver): Promise<void> {
+  debug('%O', await driver.getPaperHandlerStatus());
+}
+
+export async function setDefaults(driver: PaperHandlerDriver): Promise<void> {
+  await driver.initializePrinter();
+  debug('initialized printer (0x1B 0x40)');
+  await driver.setLineSpacing(0);
+  debug('set line spacing to 0');
+  await driver.setPrintingSpeed('slow');
+  debug('set printing speed to slow');
+}
+
+export async function printBallot(
+  driver: PaperHandlerDriver,
+  pdfData: Uint8Array,
+  options: Partial<ImageConversionOptions> = {}
+): Promise<void> {
+  const enablePrintPromise = driver.enablePrint();
+
+  let time = Date.now();
+  const pages: ImageData[] = [];
+  for await (const { page, pageCount } of pdfToImages(Buffer.from(pdfData), {
+    scale: 200 / 72,
+  })) {
+    assert(pageCount === 1, `Unexpected page count ${pageCount}`);
+    pages.push(page);
+  }
+  const page = pages[0];
+  assert(page, 'Unexpected undefined page');
+  debug(`pdf to image took ${Date.now() - time} ms`);
```

Comment:
Consider using `time` from `@votingworks/utils`.

Reply:
Removed timing info since we don't do anything with that info. Can always re add it https://github.com/votingworks/vxsuite/pull/3756/commits/736f6f48826c785778fe1d8096878def667c56a4